### PR TITLE
op-program: Compile op-program for Cannon64

### PR DESCRIPTION
**Pull request by mbaxter**
**Description**
Update `reproducible-prestate` and `op-program-client-mips` tasks to include 64-bit MTCannon targets. 

Also:
- Make a few modifications to Dockerfile.repro:
  - Don't make and then copy op-program back to the host
  - Generate meta files and copy them back to the host

### CODE REVIEW

File: op-program/Dockerfile.repro [hunk 1]
```diff
@@ -45,13 +44,21 @@ RUN mv /app/op-program/bin/0.json /app/op-program/bin/prestate-proof.json
...
 RUN mv /app/op-program/bin/0-mt.json /app/op-program/bin/prestate-proof-mt.json
 
+RUN /app/cannon/bin/cannon run --proof-at '=0' --stop-at '=1' --input /app/op-program/bin/prestate-mt64.bin.gz --meta "" --proof-fmt '/app/op-program/bin/%d-mt64.json' --output ""
+RUN mv /app/op-program/bin/0-mt64.json /app/op-program/bin/prestate-proof-mt64.json
+
 # Exports files to the specified output location.
 # Writing files to host requires buildkit to be enabled.
 # e.g. `BUILDKIT=1 docker build ...`
 FROM scratch AS export-stage
-COPY --from=builder /app/op-program/bin/op-program .
```
Comment (mbaxter) [inline op-program/Dockerfile.repro:52]: I'm not sure if there's a reason we need this?  It overwrites my local version of op-program which causes errors when I try to run it.

File: op-program/Makefile [hunk 2]
```diff
@@ -40,6 +47,8 @@ reproducible-prestate:
 	@cat ./bin/prestate-proof.json | jq -r .pre
 	@echo "MT-Cannon Absolute prestate hash: "
 	@cat ./bin/prestate-proof-mt.json | jq -r .pre
+	@echo "Cannon64 Absolute prestate hash: "
+	@cat ./bin/prestate-proof-mt64.json | jq -r .pre
```
Comment (Inphi) [inline op-program/Makefile:51]: We should hold off on doing this. The 64-bit prestate will be changed by https://github.com/ethereum-optimism/optimism/pull/12569 and once it does it'll break the reproducible-prestate job.

**Review by Inphi — COMMENTED** [review]
The changes looks good. But we should avoid checking the 64-bit prestate for reproducibility.

File: op-program/Makefile (hunk 2 above)
Comment (Inphi) [reply op-program/Makefile:51]: Good to merge this now that #12569 is merged.

**Review by Inphi — APPROVED** [review]

File: op-program/Dockerfile.repro (hunk 1 above)
Comment (mbaxter) [reply op-program/Dockerfile.repro:52]: Discussed offline and agreed this is unnecessary.

**Pull request merged**